src/Util/TestResultCache.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PHPUnit\Runner;
+
+use PHPUnit\Framework\ResultCacheListener;
+
+class TestResultCache
+{
+    /**
+     * @var string
+     */
+    public const DEFAULT_RESULT_CACHE_FILENAME = 'result_cache.json';
+
+    /**
+     * @var array
+     */
+    private $defects = [];
+
+    /**
+     * @var array
+     */
+    private $times = [];
```

Comment:
what is the content of those collections? array of.... ints? is key meaningful or numeric?
how single defect looks like ?

Reply:
Thanks, all good points regarding the results cache. Started refactoring that part to make it more robust and cleaner.